src/core/Grappa.sol

```diff
@@ -190,137 +192,99 @@ contract Grappa is OwnableUpgradeable, ReentrancyGuardUpgradeable, UUPSUpgradeab
      * @param _productId if of the product
      * @param _expiry timestamp of option expiry
-     * @param _longStrike strike price of the long option, with 6 decimals
-     * @param _shortStrike strike price of the short (upper bond for call and lower bond for put) if this is a spread. 6 decimals
+     * @param _strike strike price of the option, with 6 decimals
+     * @param _reserved allocated space for additional data
      */
-    function getTokenId(TokenType _tokenType, uint40 _productId, uint256 _expiry, uint256 _longStrike, uint256 _shortStrike)
-        external
-        pure
-        returns (uint256 id)
-    {
-        id = TokenIdUtil.getTokenId(_tokenType, _productId, uint64(_expiry), uint64(_longStrike), uint64(_shortStrike));
+    function getTokenId(
+        TokenType _optionType,
+        SettlementType _settlementType,
+        uint40 _productId,
+        uint256 _expiry,
+        uint256 _strike,
+        uint256 _reserved
+    ) external pure returns (uint256 id) {
+        id = TokenIdUtil.getTokenId(_optionType, _settlementType, _productId, uint64(_expiry), uint64(_strike), uint64(_reserved));
     }
 
     /**
-     * @notice burn option token and get out cash value at expiry
+     * @notice burn option token and settles debt and payout at expiry
      *
      * @param _account  who to settle for
      * @param _tokenId  tokenId of option token to burn
      * @param _amount   amount to settle
+     * @return debt amount owed
+     * @return payout amount receiving
      */
-    function settleOption(address _account, uint256 _tokenId, uint256 _amount) external nonReentrant returns (uint256) {
-        (address engine, address collateral, uint256 payout) = getPayout(_tokenId, _amount.toUint64());
-
-        emit OptionSettled(_account, _tokenId, _amount, payout);
-
+    function settle(address _account, uint256 _tokenId, uint256 _amount)
+        public
+        nonReentrant
+        returns (uint256 debt, uint256 payout)
+    {
         optionToken.burnGrappaOnly(_account, _tokenId, _amount);
 
-        IMarginEngine(engine).payCashValue(collateral, _account, payout);
+        Settlement memory settlement = _settle(_account, _tokenId, _amount, false);
 
-        return payout;
+        return (settlement.debt, settlement.payout);
     }
 
     /**
-     * @notice burn array of option tokens and get out cash value at expiry
+     * @notice burn array of option tokens and settles debts and payouts at expiry
      *
      * @param _account who to settle for
      * @param _tokenIds array of tokenIds to burn
-     * @param _amounts   array of amounts to burn
+     * @param _amounts array of amounts to burn
+     * @param _dryRun flag to simulate transaction
      */
-    function batchSettleOptions(address _account, uint256[] memory _tokenIds, uint256[] memory _amounts)
+    function batchSettle(address _account, uint256[] memory _tokenIds, uint256[] memory _amounts, bool _dryRun)
```

Comment:
This is a total replacement for `batchGetPayouts`. Its actually a very clean way for a subAccount to settle shorts without needing a duplicate function that essentially does the same thing (as it was previously).

Reply:
I see, but disagree this is clean. 
if you purpose is to reuse logic, a standard way would be creating a different view function that return `settlement` instead of calling a "state updating function": `_settle` and let it return the settlement struct without actually settle the option.

All you do in this function while dryRun == true, is basically to get your `debt` and `payouts`. you should create a separate function just to get that information is it's helpful.

so for `batchSettle`: you call `_settle` in the loop, and also burn token.

in `batchGetPayouts`: you call a `getSettlement` directly in the loop, add to return data.

Even if you do it this way, it won't replace `batchGetPayouts`, because you cannot send a RPC request to estimate and get the "returned data" on a state changing function.

Reply:
I can buy the integration argument!

I'll make the changes.